[USING] rework change coins (#737)
zkSNACKs/WasabiDoc: docs/using-wasabi/ChangeCoins.md

Let's assume you want to send 0.5 bitcoin to Alice.
In the `Send` tab you select one or more UTXO, these are the inputs of the transaction.

> For example, let's assume that you choose a <img src="/ShieldRed.png" alt="red shield" class="shield" /> anonymity set 1 UTXO worth 2 bitcoins.

Reviewer: ```suggestion
For example, let's assume that you choose a <img src="/ShieldRed.png" alt="red shield" class="shield" /> 1 anonymity set UTXO worth 2 bitcoins.
```

You put Alice's address in the `Receiving Address` field, and set the spending `Amount` to 0.5 bitcoin.
This will be one output of the transaction.

> Since your chosen UTXO contains 2 bitcoins, after sending 0.5 bitcoin to Alice there will be 1.5 bitcoins remaining from the original amount.

Reviewer 2: ```suggestion
Since your chosen UTXO is worth 2 bitcoins, after sending 0.5 bitcoin to Alice there will be 1.5 bitcoins change from the original amount.
```

The larger UTXO will have a colored shield (<img src="/ShieldYellow.png" alt="yellow" class="shield" />, <img src="/ShieldGreen.png" alt="green" class="shield" /> or <img src="/ShieldCheckmark.png" alt="checkmark" class="shield" /> based on your [Privacy Level Settings](/FAQ/FAQ-UseWasabi.md#how-can-i-change-the-anonset-target)) that indicates the size of the anonymity set achieved in that CoinJoin round.
The other, much smaller UTXO will be the amount (change) that was left over, and it will have a <img src="/ShieldRed.png" alt="red-shield" class="shield" /> indicating an anonymity set of `1` as it does not have the equal value denomination of the CoinJoin.

> Remember, to have an anonymity set UTXOs, all denomination outputs of a CoinJoin round must be equal.

Reviewer 2: Should we say `all` or `some` here?
```suggestion
Remember, to have anonymity set UTXOs, all denomination outputs of a CoinJoin round must be equal.
```

> For non-equal change outputs, an attacker can do subset sum analysis and find a link between input and change output.

Reviewer 2: ```suggestion
For non-equal change outputs, an attacker can do subset sum analysis and find a link between inputs and change outputs.
```

In some cases, when the change is close to the minimum denomination, this calculation can become computationally infeasible, meaning that there are too many possible interpretations that cannot be calculated.
Thus, in practice, such a change coin can have privacy too.

> But in other cases, especially when coinjoining large amounts, this link can be done easily.

Reviewer 2: ```suggestion
But in most cases, especially when coinjoining large amounts, this link can be done easily.
```

Whenever you are merging coins in one transaction, it becomes clear to any outside observer that these coins belong to the same entity, thus linking the previous transaction history.
Thus [identifying change](/why-wasabi/Coins.md#heuristics-identifying-change) based on some heuristics is a top goal of transaction surveillance.

> You want to avoid merging coins with different anonymity set values whenever possible, because this will link these coins, reduce their anonymity set to the lowest value.

Reviewer 2: ```suggestion
You want to avoid merging coins with different anonymity set values whenever possible, because this will link these coins and reduce their anonymity set to the lowest value.
```

Consider supporting invaluable projects like [The Tor Project](https://donate.torproject.org/cryptocurrency) or [the Electronic Frontier Foundation](https://supporters.eff.org/donate/donate).
You can find a list of organizations that accept Bitcoin donations [here](https://en.bitcoin.it/wiki/Donation-accepting_organizations_and_projects).

### Make it difficult to deduce the change.

There are [common heuristics identifying change outputs](/why-wasabi/Coins.md#heuristics-identifying-change), try to avoid these with every transaction.

> This means, don't reuse addresses, don't send precise amounts but randomize it, don't use replace by fee, and try to send to bech32 addresses.

Reviewer 2: ```suggestion
Don't reuse addresses, don't send precise amounts but randomize them, don't use replace by fee, and try to send to bech32 addresses.
```

After the balance of the channel is entirely on the other side, cooperatively close the channel with your peer, so that he gets the only output in the closing transaction.

Since Wasabi does not yet support Lightning Network functionality, you must use a different wallet for this task.

### Atomic swap into Lightning Network

There are some services that provide an atomic swap where you send the whole change coin to a multisignature hashed time locked contract on-chain.
In exchange, you receive a payment routed through the Lightning Network into one of your payment channels.

The swap is atomic, meaning either you receive the lightning payment, or you get the bitcoin back on-chain, so the service provider cannot steal from you.
If you use a regular Lightning invoice to receive the funds, then the service provider knows your node's public key and the channel he sends the bitcoin to.

For much better privacy, use rendezvous routing so that the sender does not gain knowledge of your receiving node.
Also ensure that the communication with the swap server is done over the TPresumably, your mother and father won't

> or network.

Reviewer 2: ???

Author: Hahaha - that's a fun copy & paste messup. 

```suggestion
Also ensure that the communication with the swap server is done over the Tor network.
```

### Consolidate several change coins, but in a CoinJoin directly.

If you consolidate many change coins in a regular non-CoinJoin transaction in the `Send` tab, then any outside observer can easily see that one user controls all these coins.
But when consolidating in a CoinJoin directly, because there are hundreds of randomly ordered inputs in a Wasabi CoinJoin transaction, it is no longer easy to find out which coins belong to one single user.

> However, a well resourced attacker can perform [subset sum analysis](/FAQ/FAQ-GeneralBitcoinPrivacy.md#what-is-a-coinjoin-sudoku) to find out which of the many inputs fund the change output.

Reviewer 2: Why are we replacing CoinJoin sudoku with subset sum analysis???

Author: Because I think it is the more adequate term. CoinJoin sudoku was a paper describing how to use subset sum analysis on CoinJoin...

Reviewer 2: I  don't agree with this. we shouldn't complicate things, CoinJoin sudoku was used heavily and it is easy to understand than subset sum analysis.
